sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -957,6 +965,15 @@ extended either for the lifetime of the graph (including both modifiable graphs
 and the executable graphs created from them) or until the buffer is no longer
 required by the graph (such as after being replaced through executable graph update).
 
+USM allocations are accessible from all nodes that are located after the malloc
+and before an (optional) free node. When the execution is in process accessing 
+the memory from anywhere else inside and outside the graph is undefined behavior.
+If the memory is not freed inside a graph, it's the users resposiblity to release 
+the memory. In that case the memory is accessible after the first execution of
+a `command_graph`. When freed inside a graph, all memory is owned by and associated
+to the lifetime of one specific `command_graph<state::executable>` object.
```

Comment:
This seems a bit error prone in that users could add allocation nodes but never free the memory. I assume the intention is so that a graph can write some values to a shared allocation which is visible on the host as well, and since the memory isn't freed by the graph it can be accessed after execution on the host. However at that point wouldn't it make more sense for the user to just allocate the memory themselves outside of the graph and handle both allocation and freeing? Alternatively if we support memcpy as a node of the graph they can copy out to some memory that they manage themselves on the host.

I think the wording is also confusing because if there is a free node inside the graph the lifetime of the memory is tied to the execution of the graph not the executable object itself. In practice yes we'd likely not actually do a free to reuse the memory but this wording seems to guarantee that the memory lives until the destruction of the graph itself. This could lead users to think they can access the memory as long as the graph is alive.

Reply:
A note that from [cudaGraphAddMemAllocNode ](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__GRAPH.html#group__CUDART__GRAPH_1g6fc8b917f0d8047f573c786d9465185d) that there is similar behaviour in adding alloc nodes that don't need to be freed in the graph 

> If the allocation is not freed in the same graph, then it can be accessed not only by nodes in the graph which are ordered after the allocation node, but also by stream operations ordered after the graph's execution but before the allocation is freed.

Reply:
That's a fair point, I think the wording could still be improved in this case though.